src/main/java/org/apache/sysml/hops/UnaryOp.java

```diff
@@ -563,6 +569,10 @@ protected double computeIntermediateMemEstimate( long dim1, long dim2, long nnz
 			// getMemEstimate works for both cases of known dims and worst-case stats
 			ret = getInput().get(0).getMemEstimate() * 3; 
 		}
+
+		if (DMLScript.USE_ACCELERATOR) {
+			OptimizerUtils.estimateSize(dim1, dim2); // Intermediate memory required to convert sparse to dense
+		}
 		
 		return ret;
 	}
```

Comment:
For matrix multiplication, intermediate memory is needed when one of the matrices is sparse and the other one dense. cuDNN is used only for dense matrices. If the input matrix is sparse, it is converted to dense.